From 585e1ead26b87f3f522cb79732c975e3b5998df9 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Fri, 14 Aug 2020 02:39:53 -0700
Subject: [PATCH 1516/1851] dmaengine: xilinx: frmbuf: Fix Y10 v4l2 pixel
 format

commit 5ffcd20fb5dc2e4cee055f0fbe157bc361fa5547 from
https://github.com/Xilinx/linux-xlnx.git

The Y10 (Luma only) pixel format is incorrectly set as V4L2_PIX_FMT_Y10
which stores 1 pixel info in 2 bytes whereas the Framebuffer Write
stores 3 pixel info in 4 bytes. So correct the V4L2 pixel format to
V4L2_PIX_FMT_XY10.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/dma/xilinx/xilinx_frmbuf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/xilinx/xilinx_frmbuf.c b/drivers/dma/xilinx/xilinx_frmbuf.c
index 54dc2c342f90..b6766da4509e 100644
--- a/drivers/dma/xilinx/xilinx_frmbuf.c
+++ b/drivers/dma/xilinx/xilinx_frmbuf.c
@@ -406,7 +406,7 @@ static const struct xilinx_frmbuf_format_desc xilinx_frmbuf_formats[] = {
 		.ppw = 3,
 		.num_planes = 1,
 		.drm_fmt = DRM_FORMAT_Y10,
-		.v4l2_fmt = V4L2_PIX_FMT_Y10,
+		.v4l2_fmt = V4L2_PIX_FMT_XY10,
 		.fmt_bitmask = BIT(17),
 	},
 	{
-- 
2.31.1

